refactor: remove DbCatalog newtype wrapper

tustvold opened the pull request:
This was a temporary hack, now that #1057 is merged we can remove it :grin:

tustvold commented on server/src/db.rs:
```diff
@@ -513,16 +493,7 @@ impl Database for Db {
-            .collect();
-
-        Ok(partition_keys)
+        Ok(self.catalog.partition_keys())
```
This has been pushed into catalog, to allow it to be used in the SchemaProvider implementation. It also is possible to implement it more efficiently, without having to lock the partitions individually.

alamb commented on server/src/db/catalog.rs:
```diff
@@ -117,6 +133,72 @@ impl Catalog {
     }
+
+    /// Return the chunks for specific partition
+    pub fn chunks(&self, partition_key: &str) -> Vec<Arc<DBChunk>> {
```
```suggestion
    pub fn db_chunks(&self, partition_key: &str) -> Vec<Arc<DBChunk>> {
```

I think this may be confusing as a `DBChunk` is not the same as a `catalog::Chunk`

I personally think this function might be better if left on `Db` so as to avoid the use of `DBChunk` at all inside the catalog. However, that is just a stylistic preference I see no practical reason that this structure is a problem

tustvold replied:
> I personally think this function might be better if left on Db so as to avoid the use of DBChunk

How do you feel about the change in general then, given the SchemaProvider gives out DBChunks?

alamb reviewed (approved):
Looks good to me

I do think it is worth considering leaving `DBChunk` separate from the `Catalog` but I don't think that would stop this PR from merging

carols10cents commented on server/src/db/catalog.rs:
```diff
@@ -117,6 +133,72 @@ impl Catalog {
+            })
+        });
+
+        names.into_iter().collect::<Vec<_>>()
```
I know you're moving this code, not adding this as new, but while we're changing it... this turbofish isn't necessary with the function's return type so this can be:

```suggestion
        names.into_iter().collect()
```

carols10cents reviewed (approved):
One little suggestion you can take or leave. Overall code looks good to me!

tustvold commented:
@alamb I've added a commit removing DBChunk from the public API of Catalog, even if it does still use it internally within SchemaProvider - let me know what you think.

I have added table methods to the catalog chunk, I imagine it is pretty much inevitable that tables will become a first-class catalog construct given there will be a parquet file per table, and so I think this makes sense longer-term

alamb commented on server/src/db.rs:
```diff
@@ -99,7 +92,7 @@ pub enum Error {
 
     #[snafu(display("Cannot write to this database: no mutable buffer configured"))]
-    DatatbaseNotWriteable {},
+    DatabaseNotWriteable {},
```
🤚 I'll take responsibility for that one

alamb commented:
@tustvold I think it looks great to me
